### PR TITLE
PAAPI: add support for protected audience extensions and "direct" buyers (`igb`)

### PR DESCRIPTION
**Pull request by dgirardi**
## Type of change

- [x] Feature

## Description of change

This adds support for the ORTB [protected audience extensions](https://github.com/InteractiveAdvertisingBureau/openrtb/blob/main/extensions/community_extensions/Protected%20Audience%20Support.md):

- when PAAPI is enabled, bid requests' `ortb2Imp.ext.igs` is populated (similar and in addition to `ext.ae`)
- bid adapters can reply with buyers (`igb`) objects in additon to auction configs (`config`). The `fledgeAuctionConfigs` response property is migrated to `paapi`. for example:
    ```
      // current response from interpetResponse, returning an auction config
     return {bids: [...], fledgeAuctionConfigs: [{bidId: 'xxx', config: {...}}]}

     // equivalent to the above, using `paapi`
     return {bids: [...], paapi: [{bidId: 'xxx', config: {...}}]}

     // return a buyer (igb) object instead of a full auction config
     return {bids: [...], paapi: [{bidId: 'xxx', igb: {...}}]}
    ```
- ortbConverter can now parse responses' `ext.igi` to generate a `paapi` array.
- publishers can set up auctions to run igb with:
  ```
   pbjs.setConfig({
      paapi: {
         enabled: true,
         componentSeller: {
            auctionConfig: {
                  decisionLogicURL: 'publisher.example'
                  // ...
            }
         }
      }
   })
  ```
- if `paapi.componentSeller.auctionConfig` is set, we attempt to collect all IGB objects into a single auction config, which is then treated like any other auction config for the purpose of `getPAAPIConfig/setPAAPIConfigForGPT`. 
- it's possible for IGB objects to be incompatible with each other (same origin, different signals). In this case we split the conflicts into a second (or third, etc) auction config.
- Because IGB objects come from different adapters, and first-party data (`ortb2`/`ortb2Imp`) can vary by adapter, auction configs generated from IGB have it set in `perBuyerSignals` instead of `auctionSignals`. "Normal" auction configs also have first party data repeated in `perBuyerSignals`, for consistency.
- Instead of one auction config (minus conflicts), it's possible to generate one for each bidder with
    ```
     pbjs.setConfig({
         paapi: {
             enabled: true,
             componentBuyers: {
                  separateAuctions: true,
                  auctionConfig: {/* ... */}
             }
         }
     })
    ```
  In this case, `auctionSignals` is populated with first party data as usual.

## Other information
Closes https://github.com/prebid/Prebid.js/issues/11131
Closes https://github.com/prebid/Prebid.js/issues/11094

### CODE REVIEW

**Comment by dgirardi** [comment]
Summary of PMC feedback:

 - `componentBuyers` config should be renamed to `componentSeller`
 - bidders should get a signal that tells them whether the publisher can set up a component seller (`bidRequest.paapi.componentSeller` ?)

**Comment by rdgordon-index** [comment]
> when PAAPI is enabled, bid requests' ortb2Imp.ext.igi is populated (similar and in addition to ext.ae)

Shouldn't this be `BidRequest.imp.ext.igs` ?

**Comment by dgirardi** [comment]
> Shouldn't this be BidRequest.imp.ext.igs ?

Yes. It looks like I used the right field in code but not the description.

**Comment by pm-harshad-mane** [comment]
@jlquaccia please review the PR

File: modules/paapi.js [hunk 1]
```diff
@@ -197,9 +315,30 @@ export function markForFledge(next, bidderRequests) {
...
+          const extAe = bidReq.ortb2Imp?.ext?.ae;
+          if (igsAe !== extAe && igsAe != null && extAe != null) {
+            logWarn(MODULE, `Bid request defines conflicting ortb2Imp.ext.ae and ortb2Imp.ext.igs, using the latter`, bidReq);
+          }
+          const bidAe = igsAe ?? extAe ?? ae;
+          if (bidAe) {
+            deepSetValue(bidReq, 'ortb2Imp.ext.ae', bidAe);
+            bidReq.ortb2Imp.ext.igs = Object.assign({
+              ae: bidAe,
+              biddable: 1
```
Comment (jlquaccia) [inline modules/paapi.js:339]: Noticed on https://github.com/InteractiveAdvertisingBureau/openrtb/blob/main/extensions/community_extensions/Protected%20Audience%20Support.md, it states that the biddable field is 0 by default (indicating a buyer is not allowed).  Also saw in the prebid "fledge" slack channel (about a month back) that the biddable field default may get reversed to 1 instead.  Is there an update on the latest around this?

Comment (dgirardi) [reply modules/paapi.js:339]: AFAIK there's no concrete plan for the spec to change.

**Review by patmmccann — APPROVED** [review]

**Pull request merged**